packages/typescript-estree/tests/lib/warn-on-unsupported-ts.test.ts

```diff
@@ -1,12 +1,18 @@
 import semver from 'semver';
 
-import * as parser from '../../src/parser';
+import type * as Parser from '../../src/parser';
 
 jest.mock('semver');
 
 const resetIsTTY = process.stdout.isTTY;
 
 describe('Warn on unsupported TypeScript version', () => {
+  let parser: typeof Parser;
+
+  beforeEach(async () => {
+    // @ts-expect-error -- We don't support ESM imports of local code yet.
+    parser = await import('../../src/parser.ts');
```

Comment:
[Cleanup] Nit: we don't generally use the `.ts` extensions, even if they are supported in the testing framework. No need.

```suggestion
    parser = await import('../../src/parser');
```

Confirming that trying to add the explicit `.js` in gives the unfortunate Jest complaint that the file can't be found.